src/fsharp/TypeChecker.fs

```diff
@@ -7635,6 +7635,11 @@ and TcComputationExpression cenv env overallTy mWhole interpExpr builderTy tpenv
         let mBuilderVal = mBuilderVal.MakeSynthetic()
         SynExpr.Lambda (false,false,SynSimplePats.SimplePats ([mkSynSimplePatVar false (mkSynId mBuilderVal builderValName)],mBuilderVal), runExpr, mBuilderVal)
 
+    let env =
+        match comp with     
+        | SynExpr.YieldOrReturn _ -> { env with eContextInfo = ContextInfo.ReturnInComputationExpression }
```

Comment:
I leave 
![image](https://cloud.githubusercontent.com/assets/57396/15358698/ef257ad4-1d05-11e6-8743-400b4247f3d6.png)

for another day ;-)